src/Kunstmaan/NodeBundle/EventListener/NodeListener.php

```diff
@@ -45,10 +47,25 @@ public function __construct(AuthorizationCheckerInterface $authorizationChecker,
     public function adaptForm(AdaptFormEvent $event)
     {
         if ($event->getPage() instanceof HasNodeInterface && !$event->getPage()->isStructureNode()) {
+			$page = $event->getPage();
+			$tabPane = $event->getTabPane();
+
             if ($this->authorizationChecker->isGranted('ROLE_PERMISSIONMANAGER')) {
-                $tabPane = $event->getTabPane();
                 $tabPane->addTab(new Tab('kuma_node.tab.permissions.title', new PermissionsFormWidget($event->getPage(), $event->getNode(), $this->permissionAdmin, $this->permissionMap)));
             }
+
+			if($page instanceof PageTabInterface) {
```

Comment:
This block is completely unrelated to the other block. In my opinion this should be split into two separate listeners to keep single responsibility. 

Then, if you have a separate listener, I see no reason for it to be in the core, you can as easily create a bundle for it and use in whichever project you’d like (even with older CMS versions)

Reply:
Thanks for your feedback. I'm a bit confused by one of your lines. You mention you see no reason for it to be in the core. Do you mean you do not want it in the NodeBundle (as in, create another bundle for it) or do you not want it in Kunstmaan at all?

Reply:
I don’t see why it should be in Kunstmaan at all, as you can easily get the same result by using an external bundle, installed separately.

Reply:
One of the reasons why we offer Kunstmaan to our customers is because it's clear and user-friendly. Being able to split a form up into multiple tabs, rather than keeping all the input fields in the Content tab, would contribute greatly to giving customers a clear overview. We could include this separately, but we believe that it should be added into the core of Kunstmaan as other users might like to offer this feature as well.

Reply:
It’s just my opinion, but I’d rather see the core with a set of basic functionalities. You can publish your code on an MIT license with a two-line install instructions, so everyone can benefit. I have contributed a bunch of stuff here, but it’s mostly bug fixes or changes to the core, not new additions. If I have an idea for a feature that’s completely isolated I don’t hesitate to publish it as a separate package (and my organisation already has a dozen of those).

YMMV. And of course I am not the maintainer here, so they can have a different take on this.

Reply:
@JZuidema @mlebkowski, all new features are a great addition. @mlebkowski I get what you mean but in cases like this we prefer to have in in the core, since it's more of an option to the `FormPage` itself. In cases like our `CookieBundle` we opted to make it a seperate bundle because it's completely isolated. If every option got it's own bundle, the power of the cms get's scattered around and users would all start to implement their own version of something that already exists. It's a thin line when you're wondering to make a seperate bundle or not but if you do, please add it to the `README`so other users can also find you're contributions :)